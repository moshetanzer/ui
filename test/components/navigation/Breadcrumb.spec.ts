import { describe, it, expect } from 'vitest'
import { UBreadcrumb } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Breadcrumb', () => {
  it.each([
    ['with links', { props: { links: [{ label: 'Home', to: '/' }, { label: 'About', to: '/about' }] } }],
    ['with divider', { props: { divider: '|', links: [{ label: 'Home', to: '/' }, { label: 'About', to: '/about' }] } }],
    ['with icon', { props: { links: [{ label: 'Home', to: '/', icon: 'i-heroicons-academic-cap' }, { label: 'About', to: '/about', icon: 'i-heroicons-code-bracket-square' }] } }],
    ['with icon slot', { slots: { icon: () => 'Icon slot' }, props: { links: [{ label: 'Home', to: '/', icon: 'i-heroicons-academic-cap' }, { label: 'About', to: '/about', icon: 'i-heroicons-code-bracket-square' }] } }],
    ['with divider slot', { slots: { divider: () => 'Divider slot' }, props: { links: [{ label: 'Home', to: '/' }, { label: 'About', to: '/about' }] } }],
    ['with default slot', { slots: { default: () => 'Default slot' }, props: { links: [{ label: 'Home', to: '/' }, { label: 'About', to: '/about' }] } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UBreadcrumb.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UBreadcrumb)
    expect(html).toMatchSnapshot()
  })
})
