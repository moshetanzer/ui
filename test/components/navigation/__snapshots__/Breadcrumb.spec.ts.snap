// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumb > renders with default slot correctly 1`] = `
"<nav aria-label="Breadcrumb" class="relative">
  <ol class="flex items-center gap-x-1.5">
    <li class="flex items-center gap-x-1.5 text-gray-500 dark:text-gray-400 text-sm leading-6 min-w-0"><a class="flex items-center gap-x-1.5 group font-semibold min-w-0 hover:text-gray-700 dark:hover:text-gray-200" href="/">
        <!--v-if-->Default slot
      </a><span class="i-heroicons-chevron-right-20-solid rtl:i-heroicons-chevron-left-20-solid flex-shrink-0 w-5 h-5" role="presentation"></span></li>
    <li class="flex items-center gap-x-1.5 text-gray-500 dark:text-gray-400 text-sm leading-6 min-w-0"><a class="flex items-center gap-x-1.5 group font-semibold min-w-0 text-primary-500 dark:text-primary-400" aria-current="page" href="/about">
        <!--v-if-->Default slot
      </a>
      <!--v-if-->
    </li>
  </ol>
</nav>"
`;

exports[`Breadcrumb > renders with divider correctly 1`] = `
"<nav aria-label="Breadcrumb" class="relative">
  <ol class="flex items-center gap-x-1.5">
    <li class="flex items-center gap-x-1.5 text-gray-500 dark:text-gray-400 text-sm leading-6 min-w-0"><a class="flex items-center gap-x-1.5 group font-semibold min-w-0 hover:text-gray-700 dark:hover:text-gray-200" href="/">
        <!--v-if--><span class="block truncate">Home</span>
      </a><span role="presentation">|</span></li>
    <li class="flex items-center gap-x-1.5 text-gray-500 dark:text-gray-400 text-sm leading-6 min-w-0"><a class="flex items-center gap-x-1.5 group font-semibold min-w-0 text-primary-500 dark:text-primary-400" aria-current="page" href="/about">
        <!--v-if--><span class="block truncate">About</span>
      </a>
      <!--v-if-->
    </li>
  </ol>
</nav>"
`;

exports[`Breadcrumb > renders with divider slot correctly 1`] = `
"<nav aria-label="Breadcrumb" class="relative">
  <ol class="flex items-center gap-x-1.5">
    <li class="flex items-center gap-x-1.5 text-gray-500 dark:text-gray-400 text-sm leading-6 min-w-0"><a class="flex items-center gap-x-1.5 group font-semibold min-w-0 hover:text-gray-700 dark:hover:text-gray-200" href="/">
        <!--v-if--><span class="block truncate">Home</span>
      </a>Divider slot</li>
    <li class="flex items-center gap-x-1.5 text-gray-500 dark:text-gray-400 text-sm leading-6 min-w-0"><a class="flex items-center gap-x-1.5 group font-semibold min-w-0 text-primary-500 dark:text-primary-400" aria-current="page" href="/about">
        <!--v-if--><span class="block truncate">About</span>
      </a>
      <!--v-if-->
    </li>
  </ol>
</nav>"
`;

exports[`Breadcrumb > renders with icon correctly 1`] = `
"<nav aria-label="Breadcrumb" class="relative">
  <ol class="flex items-center gap-x-1.5">
    <li class="flex items-center gap-x-1.5 text-gray-500 dark:text-gray-400 text-sm leading-6 min-w-0"><a class="flex items-center gap-x-1.5 group font-semibold min-w-0 hover:text-gray-700 dark:hover:text-gray-200" href="/"><span class="i-heroicons-academic-cap flex-shrink-0 w-5 h-5"></span><span class="block truncate">Home</span></a><span class="i-heroicons-chevron-right-20-solid rtl:i-heroicons-chevron-left-20-solid flex-shrink-0 w-5 h-5" role="presentation"></span></li>
    <li class="flex items-center gap-x-1.5 text-gray-500 dark:text-gray-400 text-sm leading-6 min-w-0"><a class="flex items-center gap-x-1.5 group font-semibold min-w-0 text-primary-500 dark:text-primary-400" aria-current="page" href="/about"><span class="i-heroicons-code-bracket-square flex-shrink-0 w-5 h-5"></span><span class="block truncate">About</span></a>
      <!--v-if-->
    </li>
  </ol>
</nav>"
`;

exports[`Breadcrumb > renders with icon slot correctly 1`] = `
"<nav aria-label="Breadcrumb" class="relative">
  <ol class="flex items-center gap-x-1.5">
    <li class="flex items-center gap-x-1.5 text-gray-500 dark:text-gray-400 text-sm leading-6 min-w-0"><a class="flex items-center gap-x-1.5 group font-semibold min-w-0 hover:text-gray-700 dark:hover:text-gray-200" href="/">Icon slot<span class="block truncate">Home</span></a><span class="i-heroicons-chevron-right-20-solid rtl:i-heroicons-chevron-left-20-solid flex-shrink-0 w-5 h-5" role="presentation"></span></li>
    <li class="flex items-center gap-x-1.5 text-gray-500 dark:text-gray-400 text-sm leading-6 min-w-0"><a class="flex items-center gap-x-1.5 group font-semibold min-w-0 text-primary-500 dark:text-primary-400" aria-current="page" href="/about">Icon slot<span class="block truncate">About</span></a>
      <!--v-if-->
    </li>
  </ol>
</nav>"
`;

exports[`Breadcrumb > renders with links correctly 1`] = `
"<nav aria-label="Breadcrumb" class="relative">
  <ol class="flex items-center gap-x-1.5">
    <li class="flex items-center gap-x-1.5 text-gray-500 dark:text-gray-400 text-sm leading-6 min-w-0"><a class="flex items-center gap-x-1.5 group font-semibold min-w-0 hover:text-gray-700 dark:hover:text-gray-200" href="/">
        <!--v-if--><span class="block truncate">Home</span>
      </a><span class="i-heroicons-chevron-right-20-solid rtl:i-heroicons-chevron-left-20-solid flex-shrink-0 w-5 h-5" role="presentation"></span></li>
    <li class="flex items-center gap-x-1.5 text-gray-500 dark:text-gray-400 text-sm leading-6 min-w-0"><a class="flex items-center gap-x-1.5 group font-semibold min-w-0 text-primary-500 dark:text-primary-400" aria-current="page" href="/about">
        <!--v-if--><span class="block truncate">About</span>
      </a>
      <!--v-if-->
    </li>
  </ol>
</nav>"
`;
